kobo/apps/project_views/views.py

```diff
@@ -0,0 +1,149 @@
+
+    @action(detail=True, methods=['GET'])
+    def assets(self, request, uid):
+        if not user_has_view_perms(request.user, uid):
+            raise Http404
+        assets = Asset.objects.all()
+        queryset = self.filter_queryset(
+            self._get_regional_queryset(assets, uid, _type='asset')
+        )
+        return self._get_regional_response(queryset, _type='asset')
+
+    @action(
+        detail=True,
+        methods=['GET', 'POST'],
+        url_path='(?P<_type>(assets|users))/export',
+    )
+    def export(self, request, uid, _type):
+        user = request.user
+
+        if not user_has_view_perms(user, uid):
+            raise Http404
+
+        if request.method == 'GET':
+            export = ProjectViewExportTask.objects.filter(
+                user=user, data__view=uid, data__type=_type
+            ).last()
+            if not export:
+                return Response({})
+
+            file_location = serializers.FileField().to_representation(
+                export.result
+            )
+            return Response(
+                {
+                    'status': export.status,
+                    'result': request.build_absolute_uri(file_location),
+                }
+            )
+        elif request.method == 'POST':
+            export_task = ProjectViewExportTask.objects.create(
+                user=user,
+                data={
+                    'view': uid,
+                    'type': _type,
+                },
+            )
+
+            # Have Celery run the export in the background
+            project_view_export_in_background.delay(
+                export_task_uid=export_task.uid,
+                username=user.username,
+            )
+
+            return Response({'status': export_task.status})
+
+    @action(detail=True, methods=['GET'])
+    def users(self, request, uid):
+        if not user_has_view_perms(request.user, uid):
+            raise Http404
+        users = User.objects.all()
+        queryset = self.filter_queryset(
+            self._get_regional_queryset(users, uid, _type='user')
+            .exclude(pk=settings.ANONYMOUS_USER_ID)
+            .distinct()
+            .order_by('id')
+        )
+        return self._get_regional_response(queryset, _type='user')
+
+    def get_serializer_context(self):
+        context_ = super().get_serializer_context()
+        context_['request'] = self.request
+        return context_
+
+    def _get_regional_response(self, queryset, _type):
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self._get_regional_serializer(page, _type=_type)
+            return self.get_paginated_response(serializer.data)
+        serializer = self._get_regional_serializer(queryset, _type=_type)
+        return Response(serializer.data)
+
+    def _get_regional_serializer(self, queryset, _type):
+        serializer = AssetMetadataListSerializer
```

Comment:
The serializer context should contain `object_permissions_per_asset` property when serializer is `AssetMetadataListSerializer`. It optimizes the queries made to DB to retrieve permissions in assets list. 
 
See https://github.com/kobotoolbox/kpi/blob/eab99a8f6c695f4f3ab36b8e55d7394c3042925b/kpi/views/v2/asset.py#L594-L614

Obviously `AssetMetadataListSerializer` should extend  `AssetListSerializer`

Reply:
kpi#4195